src/Environment/Environment.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file was originally part of the league/commonmark package.
+ *
+ * (c) Colin O'Dell <colinodell@gmail.com>
+ *
+ * Original code based on the CommonMark JS reference parser (https://bitly.com/commonmark-js)
+ *  - (c) John MacFarlane
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace UnicornFail\Emoji\Environment;
+
+use UnicornFail\Emoji\Dataset\Dataset;
+use UnicornFail\Emoji\Dataset\DatasetInterface;
+use UnicornFail\Emoji\Emojibase\ShortcodeInterface;
+use UnicornFail\Emoji\Exception\LocalePresetException;
+use UnicornFail\Emoji\Extension\CoreExtension;
+use UnicornFail\Emoji\Parser\Parser;
+use UnicornFail\Emoji\Parser\ParserInterface;
+
+final class Environment extends AbstractConfigurableEnvironment implements EmojiEnvironmentInterface
```

Comment:
Clearly commonmark will have very commonmark specific requirements, but that can easily be addressed in a `CommonMarkEnvironmentInterface` as I did here with `EmojiEnvironmentInterface`.

Reply:
This has been simplified.